fusestore/store/src/executor/meta_handlers.rs

```diff
@@ -23,40 +22,63 @@ use common_flights::meta_api_impl::GetDatabaseAction;
 use common_flights::meta_api_impl::GetDatabaseActionResult;
 use common_flights::meta_api_impl::GetTableAction;
 use common_flights::meta_api_impl::GetTableActionResult;
+use common_metatypes::Table;
 use log::info;
 
 use crate::executor::action_handler::RequestHandler;
 use crate::executor::ActionHandler;
-use crate::protobuf::CmdCreateDatabase;
-use crate::protobuf::CmdCreateTable;
-use crate::protobuf::Db;
-use crate::protobuf::Table;
+use crate::meta_service::cmd::Cmd::CreateDatabase;
+use crate::meta_service::cmd::Cmd::CreateTable;
+use crate::meta_service::cmd::Cmd::DropDatabase;
+use crate::meta_service::cmd::Cmd::DropTable;
+use crate::meta_service::AppliedState;
+use crate::meta_service::LogEntry;
 
 // Db
-
 #[async_trait::async_trait]
 impl RequestHandler<CreateDatabaseAction> for ActionHandler {
     async fn handle(
         &self,
         act: CreateDatabaseAction,
     ) -> common_exception::Result<CreateDatabaseActionResult> {
         let plan = act.plan;
-        let mut meta = self.meta.lock();
-
-        let cmd = CmdCreateDatabase {
-            db_name: plan.db,
-            db: Some(Db {
-                // meta fills it
-                db_id: -1,
-                ver: -1,
-                table_name_to_id: HashMap::new(),
-                tables: HashMap::new(),
-            }),
+        let db_name = &plan.db;
+        let if_not_exists = plan.if_not_exists;
+
+        let db = self.meta_node.get_database(db_name).await;
+        if let Some(db) = db {
+            return if if_not_exists {
+                Ok(CreateDatabaseActionResult {
+                    database_id: db.database_id,
+                })
+            } else {
+                Err(ErrorCode::DatabaseAlreadyExists(format!(
+                    "{} database exists",
+                    db_name
+                )))
+            };
+        }
```

Comment:
Reading from `meta_node` does not guarantee it sees the latest state, e.g., the database is already deleted(a user already received a positive ack of a delete operation and then at once sent a create-db operation, which should succeed.), but the change has not yet been replicated to this node.

Thus this checking should always be done at the apply phase in the state machine.